site/app/views/forum/ForumThreadView.php

```diff
@@ -590,162 +590,150 @@ public function showAlteredDisplayList($threads, $filtering, $thread_id, $catego
 		return $this->displayThreadList($threads, $filtering, $threadAnnouncement, $activeThreadTitle, $tempArray, $thread_id, $categories_ids);
 	}
 
-	public function displayThreadList($threads, $filtering, &$activeThreadAnnouncement, &$activeThreadTitle, &$activeThread, $thread_id_p, $current_categories_ids){
-					$return = "";
-					$used_active = false; //used for the first one if there is not thread_id set
-					$current_user = $this->core->getUser()->getId();
-					$display_thread_ids = $this->core->getUser()->getGroup() <= 2;
-
-					$activeThreadAnnouncement = false;
-					$activeThreadTitle = "";
-					$function_date = 'date_format';
-					$activeThread = array();
-
-					foreach($threads as $thread){
-						$first_post = $this->core->getQueries()->getFirstPostForThread($thread["id"]);
-						if(is_null($first_post)) {
-							// Thread without any posts(eg. Merged Thread)
-							$first_post = array('content' => "");
-							$date = null;
-						} else {
-							$date = DateUtils::parseDateTime($first_post['timestamp'], $this->core->getConfig()->getTimezone());
-						}
-						if($thread['merged_thread_id'] != -1){
-							// For the merged threads
-							$thread['status'] = 0;
-						}
+    public function displayThreadList($threads, $filtering, &$activeThreadAnnouncement, &$activeThreadTitle, &$activeThread, $thread_id_p, $current_categories_ids)
+    {
+        $return = "";
+        $used_active = false; //used for the first one if there is not thread_id set
+        $current_user = $this->core->getUser()->getId();
+        $display_thread_ids = $this->core->getUser()->getGroup() <= 2;
```

Comment:
Let's change this line to: `$display_thread_ids = $this->core->getUser()->accessFullGrading()`